src/os/bluestore/BlueStore.cc

```diff
@@ -13623,6 +13691,15 @@ int BlueStore::_do_alloc_write(
   }
   _collect_allocation_stats(need, min_alloc_size, prealloc.size());
 
+  if (bdev->is_smr()) {
+    std::lock_guard l{zoned_cleaner_lock};
+    if (!zoned_cleaner_in_progress && shared_alloc.a->zoned_needs_cleaning()) {
```

Comment:
Wouldn't it better call zoned_need_cleaning before acquiring zoned_cleaner_lock?

Reply:
I tried to follow the pattern of similar existing threads (such as kv_finalize_thread), but sure, we can do it like you say.  I've changed the code as you've suggested.